game-core/src/main/java/games/strategy/triplea/attachments/AbstractConditionsAttachment.java

```diff
@@ -375,4 +399,50 @@ public void changeChanceDecrementOrIncrementOnSuccessOrFailure(final IDelegateBr
       delegateBridge.addChange(ChangeFactory.attachmentPropertyChange(this, newChance, CHANCE));
     }
   }
+
+  private static Map<String, Function<IAttachment, AttachmentProperty<?>>> getPopulatedAttachmentMap() {
+    return ImmutableMap.<String, Function<IAttachment, AttachmentProperty<?>>>builder()
+        .put("conditions",
+            ofCast(a -> AttachmentProperty.of(
+                a::setConditions,
+                a::setConditions,
+                a::getConditions,
+                a::resetConditions)))
+        .put("conditionType",
+            ofCast(a -> AttachmentProperty.of(
+                a::setConditionType,
+                a::setConditionType,
+                a::getConditionType,
+                a::resetConditionType)))
+        .put("invert",
+            ofCast(a -> AttachmentProperty.of(
+                a::setInvert,
+                a::setInvert,
+                a::getInvert,
+                a::resetInvert)))
+        .put("chance",
+            ofCast(a -> AttachmentProperty.of(
+                a::setChance,
+                a::setChance,
+                a::getChance,
+                a::resetChance)))
+        .put("chanceIncrementOnFailure",
+            ofCast(a -> AttachmentProperty.of(
+                a::setChanceIncrementOnFailure,
+                a::setChanceIncrementOnFailure,
+                a::getChanceIncrementOnFailure,
+                a::resetChanceIncrementOnFailure)))
+        .put("chanceDecrementOnSuccess",
```

Comment:
I suspect each of these strings should be attached to an enum, which would be a good place to have the casting mapping as well.